src/NGen.cpp

```diff
@@ -126,6 +130,22 @@ int main(int argc, char *argv[]) {
         MPI_Comm_size(MPI_COMM_WORLD, &mpi_num_procs);
         #endif // NGEN_MPI_ACTIVE
 
+        #ifdef WRITE_PID_FILE_FOR_GDB_SERVER
```

Comment:
Do we want to keep this in master?

Reply:
@robertbartel what are your thoughts?  When I first seen it, I thought about putting hiding some of this in a function in a utility header and then using a `DEBUG LEVEL` macro to call that function if needed.

Reply:
My thought is leave it in.  First, it's a very small amount of code being added (or not).  Second, the purpose was to add support for "remote" debugging by way of a Docker container, to hopefully better be able to step through the Fortran code.  The setup just wasn't finished before the issues needing debugging were otherwise resolved. Still, I think it's good to leave this in place, rather than lose the progress (the general idea came from something I started trying to debug the MPI related work).

As far as the utility function, I think this is minimal enough to just leave where it is, unless you anticipate something else needing to reuse the functionality.   I also think it should have a dedicated macro, rather than be tied directly to a debug level, for more granular control over whether this is active.

Reply:
I'm OK with it, just like to make sure anything going in the "main" source/function is thought through a bit.